### DISCUSSION TITLE
make one field update another one

### DISCUSSION THREAD

**Discussion by kamil-malinski** [directus/directus#12406] [Archive]
Hi!

I'm trying to find a way to update a field from another one in the edit-view. My use-case is that I have a title-field and a slug-field and I want the slug-field to autogenerate from the title-field as it is being entered but still let the user change the slug-field if necessary. 

Is there a way to access an interface from another interface?

Cheers

MJFox

**Comment by benhaynes** [comment]
I defer to @rijkvanzanten on this!

**CC @nickrum @rijkvanzanten on the idea below...**
It would be nice if we could enable front-end "hooks" for the Item Form within Data Flows. Would that ever be possible? There are a LOT of instances where admins could add data logic to their forms before the formal _save_ event (`items.create` and `items.update`). How difficult would it be to add the following to Data Flows?

* **Trigger**: On Field Update (in client-side app)
* **Action**: Update Field Value (in client-side app)
* **Action**: Update Field Options (in client-side app)

Those three things would enable... a LOT. There's also a good overlap with Field Conditions here, but that can be another conversation.

**Comment by rijkvanzanten** [comment]
Support for that was just added for custom interfaces in https://github.com/directus/directus/pull/11620 🙂

**Comment by kueronya** [comment]
Just stumbled across this during my flows research. 

Is the proposal from @benhaynes still on the table or was it discarded for a specific reasons?

Manipulating form/field state via flows would solve so many (often small but tedious) problems without writing a dedicated interface/display/hook.
